show/main.py

```diff
@@ -393,6 +395,44 @@ def run_command_in_alias_mode(command):
         sys.exit(rc)
 
 
+def get_bgp_summary_extended(command_output):
+    """
+    Adds Neighbor name to the show ip[v6] bgp summary command
+    :param command: command to get bgp summary
+    """
+    modified_output = []
+    my_list = iter(command_output.splitlines())
+    for element in my_list:
+        if element.startswith("Neighbor"):
+            element = "{}\tNeighborName".format(element)
+            modified_output.append(element)
+        elif not element or element.startswith("Total number "):
+            modified_output.append(element)
+        elif re.match(r"([0-9A-Fa-f]{1,4}:|\d+.\d+.\d+.\d+)", element.split()[0]):
+            ip = element.split()[0]
+            name = bgp_neighbor_ip_to_name(ip)
+            if len(element.split()) == 1:
+                modified_output.append(element)
+                element = next(my_list)
+            element = "{}\t{}".format(element, name)
+            modified_output.append(element)
+        else:
+            modified_output.append(element)
+    click.echo("\n".join(modified_output))
+
+
+def bgp_neighbor_ip_to_name(address):
+    """
+    Gets BGP neighbor name from the given ipv4/v6 address
+    :param address: ipv4/ipv6 address
+    :return: string
+    """
+    config_db = ConfigDBConnector()
+    config_db.connect()
```

Comment:
Can you open that connection only once?
Otherwise you'll open a connection for every neighbor

Reply:
changed in next iteration